railties/test/application/rake_test.rb

```diff
@@ -126,6 +126,36 @@ def test_rake_routes_with_controller_environment
       assert_equal "Prefix Verb URI Pattern     Controller#Action\n  cart GET  /cart(.:format) cart#show\n", output
     end
 
+    def test_rake_routes_with_global_search_key
```

Comment:
Can you add `assert_deprecated` to the `ENV['CONTROLLER']` test just above this test?